### PR TITLE
Add continuous counter update to VaCounter close #2108

### LINKED ISSUE #2108: VaCounter increase/decrease value on long click
For me, it's intuitive if I keydown on increase button and wait, it will automatically increase value until I keyup.

### PR DESCRIPTION
**Pull request by Skander999**
## Description
This PR adds a feature to the counter component in Vuestic UI that allows the counter to continuously update when the user keeps the button pressed instead of updating only once. close #2108

## Markup:
<details>
<summary>Relevant code</summary>

```vue
<template>
  ...
  <va-button class="va-counter__button-decrease" :aria-label="t('decreaseCounter')" v-bind="decreaseButtonProps"
    @click="decreaseCount" @mousedown.prevent="startDecreasing" @mouseup.prevent="stopDecreasing" />
  ...
  <va-button class="va-counter__button-increase" :aria-label="t('increaseCounter')" v-bind="increaseButtonProps"
    @click="increaseCount" @mousedown.prevent="startIncreasing" @mouseup.prevent="stopIncreasing" />
  ...
</template>

<script>
...
let decreaseInterval: number | undefined
let increaseInterval: number | undefined
...
const startDecreasing = () => {
  decreaseCount()
  decreaseInterval = setInterval(decreaseCount, 100) as any
}
const stopDecreasing = () => {
  clearInterval(decreaseInterval)
}
const startIncreasing = () => {
  increaseCount()
  increaseInterval = setInterval(increaseCount, 100) as any
}
const stopIncreasing = () => {
  clearInterval(increaseInterval)
}
...
setup(props, { emit, attrs }) {
  ...
  return {
    ...
    startDecreasing,
    stopDecreasing,
    startIncreasing,
    stopIncreasing,
    ...
  }
}
...
</script>


## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

**Review by RVitaly1978 — APPROVED** [review]

File: packages/ui/src/components/va-counter/VaCounter.vue
Comment (m0ksem) [inline packages/ui/src/components/va-counter/VaCounter.vue:23]: It looks like something you want to have in composable:

```ts
const { listeners } = useLongPress({
  start: () => alert(), 
  finish: () => alert(),
})
```

or using useTemplateRef to make it clearer.

```ts
useLongPress('increaseButtonRef', {
  start: () => alert(), 
  finish: () => alert(),
})
```

I'm not sure if we already have this one. Need to search for something like usePressed or useMouse.

This is the way I would do it.

**Review by m0ksem — COMMENTED** [review]
Looks good. Thanks! Will do QA later. Try to move logic outside this component to compostables. We likely will use it for other components like Carousel, Tabs, etc.

File: packages/ui/src/components/va-counter/VaCounter.vue
Comment (Skander999) [reply packages/ui/src/components/va-counter/VaCounter.vue:22]: Thank you for the feedback! I tried to follow your advice and moved the logic to useLongPress in the compostables. Please let me know if you have any other suggestions or tips for the code!

**Review by m0ksem — APPROVED** [review]

**Comment by m0ksem** [comment]
Hey, @Skander999. Composable looks good, thanks! I simplified it a bit using our existing utility composables like useEvent and useTemplateRef.

**Comment by m0ksem** [comment]
@RVitaly1978, check how your unessesary refactroing impact other's PR-  it was beautiful until I merged develop here to resolve conflicts :) Also, waiting for your review here and we're ready to merge.

**Review by RVitaly1978 — APPROVED** [review]

**Comment by Skander999** [comment]
Thank you for your feedback @m0ksem !

**Pull request merged**
